src/cascadia/TerminalApp/AppActionHandlers.cpp

```diff
@@ -1264,25 +1264,62 @@ namespace winrt::TerminalApp::implementation
         {
             if (const auto& realArgs = args.ActionArgs().try_as<SuggestionsArgs>())
             {
-                auto source = realArgs.Source();
-
-                switch (source)
-                {
-                case SuggestionsSource::CommandHistory:
+                const auto source = realArgs.Source();
+                std::vector<Command> commandsCollection;
+                Control::CommandHistoryContext context{ nullptr };
+                winrt::hstring currentCommandline = L"";
+
+                // If the user wanted to use the current commandline to filter results,
+                //    OR they wanted command history (or some other source that
+                //       requires context from the control)
+                // then get that here.
+                const bool shouldGetContext = realArgs.UseCommandline() ||
+                                              WI_IsFlagSet(source, SuggestionsSource::CommandHistory);
+                if (shouldGetContext)
                 {
                     if (const auto& control{ _GetActiveControl() })

```

Comment:
Hey, didn't we make a change on your other suggestions PR to use `sender` instead of `active control`?